gfx_window.cpp

```diff
@@ -266,6 +289,28 @@ class GfxWindowInternal
                     DragFinish(hdrop);
                 }
                 break;
+            case WM_GETDPISCALEDSIZE:
+                {
+                    DWORD const dpi = (WORD)w_param;
+                    RECT window_rect = { 0, 0, (LONG)gfx_window->window_width_, (LONG)gfx_window->window_height_ };
+                    DWORD const window_style = WS_OVERLAPPEDWINDOW & ~((gfx_window->flags_ & kGfxCreateWindowFlag_NoResizeWindow) != 0 ?
+                                                                           WS_THICKFRAME | WS_MINIMIZEBOX | WS_MAXIMIZEBOX : 0);
+                    DWORD const window_style_ex = WS_EX_OVERLAPPEDWINDOW & ((gfx_window->flags_ & kGfxCreateWindowFlag_AcceptDrop) != 0 ? WS_EX_ACCEPTFILES : 0);
+                    AdjustWindowRectExForDpi(&window_rect, window_style, FALSE, window_style_ex, dpi);
+                    SIZE* const out = (SIZE*)l_param;
+                    out->cx = window_rect.right - window_rect.left;
+                    out->cy = window_rect.bottom - window_rect.top;
+                    return TRUE;
+                }
```

Comment:
`WM_GETDPISCALEDSIZE` is sent before a window is resized by windows based on a DPI change. If left unhandled then windows will automatically resize the window based on the DPI change (i.e. if changing from 100% to 150% scale then the window size will also be increased to 150%)
With this event it is possible to intersect the window resize and specify the desired size. In this case we maintain the user set width/height essentially preventing windows from resizing the back bufer. This way a back buffer of 1920x1080 for instance will stay at the exact size even when changing DPI